src/System.Data.SqlClient/src/System.Data.SqlClient.csproj

```diff
@@ -11,16 +11,19 @@
     <PlatformTarget>AnyCPU</PlatformTarget>
     <PackageTargetFramework>dotnet5.5</PackageTargetFramework>
     <UsePackageTargetRuntimeDefaults>true</UsePackageTargetRuntimeDefaults>
+    <ProjectGuid>{D4550556-4745-457F-BA8F-3EBF3836D6B4}</ProjectGuid>
   </PropertyGroup>
   <PropertyGroup Condition=" '$(TargetsWindows)' != 'true' ">
     <DefineConstants>$(DefineConstants);MANAGED_SNI</DefineConstants>
   </PropertyGroup>
+  <PropertyGroup Condition=" '$(TargetsWindows)' == 'true' ">
```

Comment:
One way of getting rid of the #if is depicted in System.Runtime.InteropServices 

https://github.com/dotnet/corefx/blob/master/src/System.Runtime.InteropServices.RuntimeInformation/src/OSPlatform.cs

You can have the LocaleHelper.cs and have different implementations being compiled on different platforms. 
A functions of LocaleHelper will call into the plat specific implementation and that way you will not need these constants.

We want to bring down or potentially remove the usage of these Constants.